pkgs/development/libraries/qt-6/functions/broken-mkspecs-files-disable.sh

```diff
@@ -0,0 +1,53 @@
+# https://github.com/NixOS/nixpkgs/issues/214765
+
+# the functions qtPrepareTool and qtPrepareLibExecTool are defined in
+# qtbase-dev/mkspecs/features/qt_functions.prf
+
+brokenMkspecsFilesDisable() {
+  # check if a tool is installed in the expected location.
+  # if the tool is not installed, disable the mkspecs file
+  # by moving it to *.bak.
+  # the disabled mkspecs files will be restored by
+  # restoreMkspecsFiles in downstream derivations, for example qttools.
+  local source_prefix=$1 # $dev
+  local target_prefix=$2 # $dev
+  local instloc_expected=$3 # $dev/bin or $dev/libexec
+  local funcname_expected=$4 # qtPrepareTool or qtPrepareLibExecTool
+  local mkspecs_path=mkspecs/features
+  local mkspecs_file_pattern='*.prf'
+  local debug=${DEBUG:-false}
+  local dryrun=${DRYRUN:-false}
+  local dryrun_cmd=
+  if $dryrun; then dryrun_cmd=echo; fi
+  while read -r file_match_line; do
+    IFS=, read -r mkspecs_file funcname variable \
+      default_tool suffix prepare instloc \
+      < <(sed -E -e 's/\)$//' -e 's/, /,/g' -e 's/[:(]/,/g' <<<"$file_match_line")
+    local match_line=${file_match_line#*:}
+    local tool_file=$instloc_expected/$default_tool
+    if $debug; then
+      echo brokenMkspecsFilesDisable: debug:
+      echo file_match_line="$file_match_line"; echo match_line="$match_line"
+      echo mkspecs_file=$mkspecs_file; echo funcname=$funcname
+      echo variable=$variable; echo default_tool=$default_tool
+      echo suffix=$suffix; echo prepare=$prepare; echo instloc=$instloc
+      echo tool_file=$tool_file
+    fi
+    # fix: $$qmlplugindump -> qmlplugindump
+    if [[ ${default_tool:0:2} == '$$' ]]
+      then default_tool=${default_tool:2}; fi
```

Comment:
There should be some bash magic to do the strip prefix things.

Reply:
yes, `default_tool=${default_tool/'$$'/}` is shorter